src/libpmem/pmem.c

```diff
@@ -189,18 +191,23 @@
 #include "mmap.h"
 #include "file.h"
 #include "valgrind_internal.h"
+#ifdef AARCH64
+#include "neon_sse.h"
+#endif
 
 #ifndef _MSC_VER
 /*
  * The x86 memory instructions are new enough that the compiler
  * intrinsic functions are not always available.  The intrinsic
  * functions are defined here in terms of asm statements for now.
  */
+
+#ifndef AARCH64
```

Comment:
I don't really like this. Maybe we should think about architecture specific headers with these definitions?

Reply:
You mean create different headers for x86 and ARM right? I can try to create different headers in libpmem/ and move these there if you think that can help.

Reply:
Yup, we should abstract away the arch and make two implementations. memcpy will probably also need to be implemented separately for different archs (initially, we could just implement memcpy as libc memcpy() + persist())

Take a look here:
https://github.com/rlane/dpdk/tree/master/lib/librte_eal/common/include/arch

Reply:
After discussion, since the number of ARM-specific changes is fairly minimal in the latest version of this patch, this patch keeps the ifdefs instead of separate arch specific folders as suggested. This could be refactored as more arch specific changes are added IMHO.